packages/service-library/tests/test_background_task.py

```diff
@@ -127,4 +127,5 @@ async def test_periodic_task_context_manager(
         await asyncio.sleep(5 * task_interval.total_seconds())
         assert asyncio_task.cancelled() is False
         assert asyncio_task.done() is False
-    assert asyncio_task.cancelled() is True
+    # NOTE: task is no longer cancelled but gracefully stopped
+    assert asyncio_task.cancelled() is False
```

Comment:
This is a bit confusing. So your task is neither `done` nor `canceled`? Is it pending? but wait, you are outside of the context manager? What is the state of the task?